### PR TITLE
Refactor orders 81

### LINKED ISSUE #81: Refactor Orders
- submit_order() in orders controller would work better as instance method in Orders model. 
- see if we can use stubbing to set @cart in checkout feature test.

### PR DESCRIPTION
**Pull request by meyerhoferc**
adds model test for cart in order to move .submit_order() to the cart class. 
closes #81 
was not able to implement stubbing for the cart, this could be something we delve into at the end of the project

### CODE REVIEW

File: app/controllers/orders_controller.rb [hunk 1]
```diff
@@ -10,17 +9,9 @@ def show
 
   def create
     order = current_user.orders.create(total_price: @cart.total)
-    submit_order(order)
+    order.submit_order(@cart)
```
Comment (CjMoore) [inline app/controllers/orders_controller.rb:12]: 👍

**Review by CjMoore — APPROVED** [review]
Looks good. Ready for second review and merge @valerietrudell @cmacaulay

File: app/controllers/orders_controller.rb (hunk 1 above)
Comment (valerietrudell) [inline app/controllers/orders_controller.rb]: 🔥 🔥

File: spec/models/order_spec.rb [hunk 2]
```diff
@@ -14,4 +14,22 @@
...
+    cart.add_item(item2.id)
+
+    order = user.orders.create!(total_price: cart.total)
+    order.submit_order(cart)
+
+    expect(order.persisted?).to eq(true)
+    expect(order.order_items.count).to eq(2)
+    expect(OrderItem.find_by(item: item1).quantity).to eq(2)
+    expect(OrderItem.find_by(item: item2).quantity).to eq(1)
+  end
```
Comment (cmacaulay) [inline spec/models/order_spec.rb:34]: 👍

File: app/models/order.rb [hunk 3]
```diff
@@ -7,4 +7,11 @@ class Order < ApplicationRecord
   has_many :items, through: :order_items
 
   enum status: ['ordered', 'paid', 'canceled', 'completed']
+
+  def submit_order(cart)
+    cart.items.each do |item, quantity|
+      order_items.create!(item_id: item.id,
+                          quantity: quantity)
+    end
+  end
```
Comment (cmacaulay) [inline app/models/order.rb:16]: this is great!

File: app/controllers/orders_controller.rb [hunk 4]
```diff
@@ -10,17 +9,9 @@ def show
...
     session.delete(:cart)
     flash[:success] = "Mischief is coming your way, you have successfully placed your order!"
     redirect_to orders_path
   end
```
Comment (cmacaulay) [inline app/controllers/orders_controller.rb:16]: 👍 so fresh and so clean, clean.

**Review by cmacaulay — COMMENTED** [review]
Looks great!

**Pull request merged**